### PR TITLE
Implement LAST_DAY function #1315

### LINKED ISSUE #1295: Implement LAST_DAY function
# Description
Extract the last day of the month for the given date.

# Example
```
SELECT LAST_DAY("2017-06-20");
SELECT LAST_DAY("2017-02-10 09:34:00");
```

### PR DESCRIPTION
**Pull request by cjy13753**
resolve #1295

### CODE REVIEW

File: core/src/executor/evaluate/function.rs [hunk 1]
```diff
@@ -634,6 +635,24 @@ pub fn format<'a>(
...
+            let last_day = calculate_last_day(timestamp.date());
+            Ok(Evaluated::from(Value::Date(last_day)))
+        }
+        _ => Err(EvaluateError::FunctionRequiresDateOrDateTimeValue(name).into()),
+    }
+}
+
+fn calculate_last_day(date: chrono::NaiveDate) -> chrono::NaiveDate {
+    date + Months::new(1) - Duration::days(date.day() as i64)
+}
```
Comment (panarch) [inline core/src/executor/evaluate/function.rs:654]: we can try simplifying the codes a bit like this.

```rust
let date = match expr.try_into()? {
    Value::Date(date) => date,
    Value::Timestamp(timestamp) => timestamp.date(),
    _ => {
        return Err(EvaluateError::FunctionRequiresDateOrDateTimeValue(name).into());
    }
};

Ok(Evaluated::Value(Value::Date(
    date + Months::new(1) - Duration::days(date.day() as i64),
)))
```

File: core/src/ast/function.rs [hunk 2]
```diff
@@ -302,6 +303,7 @@ impl ToSql for Function {
             Function::Degrees(e) => format!("DEGREES({})", e.to_sql()),
             Function::Now() => "NOW()".to_owned(),
             Function::Pi() => "PI()".to_owned(),
+            Function::LastDay(expr) => format!("LAST_DAY({})", expr.to_sql()),
```
Comment (panarch) [inline core/src/ast/function.rs:306]: could you add an unit test for this?

File: core/src/plan/expr/function.rs [hunk 3]
```diff
@@ -45,6 +45,7 @@ impl Function {
             | Self::Ascii(expr)
             | Self::Chr(expr)
             | Self::Md5(expr)
+            | Self::LastDay(expr)
```
Comment (panarch) [inline core/src/plan/expr/function.rs:48]: this, too
could you add an unit test?

**Review by panarch — APPROVED** [review]
Looks all great! Thanks a lot for the contribution 👍 👍

**Comment by cjy13753** [comment]
This is my first ever PR to an open source project. Appreciate your help and feedback!

**Pull request merged**